src/vaadin-radio-button.html

```diff
@@ -251,7 +255,13 @@
          * @protected
          */
         click() {
-          this.shadowRoot.querySelector('input').click();
+          // If you change this block, please test manually that radio-button and
+          // radio-group still works ok on iOS 12/13 and up as it may cause
+          // an issue that is not possible to test programmatically.
+          // See: https://github.com/vaadin/vaadin-radio-button/issues/140
+          if (!this.disabled) {
+            this.shadowRoot.querySelector('input').dispatchEvent(new MouseEvent('click'));
+          }
```

Comment:
The `disabled` check is needed here now since otherwise manually dispatching a click event will cause even a disabled `<input type="radio">` to become checked.